fxa-client/src/lib.rs

```diff
@@ -498,24 +503,91 @@ impl FirefoxAccount {
         self.state.config.token_server_endpoint_url()
     }
 
-    pub fn handle_push_message(&self) {
-        panic!("Not implemented yet!")
+    /// Retrieve our own client instance metadata.
+    pub fn client_instance(&mut self) -> Result<ClientInstanceResponse> {
+        let refresh_token = self.get_refresh_token()?;
+        let client = Client::new(&self.state.config);
+        client.client_instance(&refresh_token)
+    }
+
+    /// Retrieve the metadata from all the clients instances connected to
+    /// the account (including ours).
+    pub fn clients_instances(&mut self) -> Result<Vec<ClientInstanceResponse>> {
+        let access_token = self.get_access_token(scopes::CLIENTS_READ)?.token;
+        let client = Client::new(&self.state.config);
+        client.clients_instances(&access_token)
+    }
+
+    pub fn pending_commands(&self) -> Result<PendingCommandsResponse> {
+        let refresh_token = self.get_refresh_token()?;
+        let client = Client::new(&self.state.config);
+        client.pending_commands(refresh_token)
+    }
+
+    pub fn invoke_command(
+        &mut self,
+        command: &str,
+        target_device_id: &str,
+        payload: &serde_json::Value,
+    ) -> Result<()> {
+        let access_token = self.get_access_token(scopes::COMMANDS_WRITE)?.token;
+        let client = Client::new(&self.state.config);
+        client.invoke_command(&access_token, command, target_device_id, payload)
+    }
+
+    pub fn set_push_subscription(&self, push_subscription: PushSubscription) -> Result<()> {
+        let metadata = ClientInstanceRequestBuilder::new()
+            .push_subscription(push_subscription)
+            .build();
+        self.upsert_client_instance(metadata)
+    }
+
+    pub fn set_name(&self, name: &str) -> Result<()> {
```

Comment:
Are we happy to call this just "name" or would callers of the API appreciate something more specific like "display name"?